File: 2019-2020/reverse/1/Lrep/crack.pl
#!/usr/bin/perl -l

$text = "2I7VjH)Q3RkPK*Sm3R!aY8!1P:5T6YL-`N4U5M,;xB#v1D%24U";
$input = "";
for $i (0..length($text)-2){
    ($a,$b) = (substr($text,$i,1),substr($text,$i+1,1));
    if (ord($a) == (ord($b) ^ 97)){
        $input = "$input$a";
    }
}
print $input;

File: 2019-2020/reverse/2/Venkata/crack.c
#include <stdio.h>

int main()
{
	char input[256];
	int len;
	int i;
	
	// len = 2k + 1 and len > 4
	len = 5;
	input[0] = 94;
	input[len >> 1] = 95;
	input[len - 1] = 94;
	
	for (i = 1; i < (len >> 1); i++)
	{
		input[i] = 33;
	}
	
	for (i = (len >> 1) + 1; i < len - 1; i++)
	{
		input[i] = 48;
	}	
	
	printf("%s\n", input);
	return 0;
}